src/ocaml/ml_kind.ml

```diff
@@ -54,3 +54,30 @@ module Dict = struct
     let open Dyn in
     record [ ("impl", f impl); ("intf", f intf) ]
 end
+
+module Dialect = struct
+  type 'a t =
+    { impl : 'a
+    ; intf : 'a option
```

Comment:
Is there a reason to make `impl` mandatory? Why not keep the symmetry and make both optional?

Reply:
done